src/integrations/prefect-email/prefect_email/credentials.py

```diff
@@ -182,13 +191,22 @@ def example_get_server_flow():
         if smtp_type == SMTPType.INSECURE:
             server = SMTP(smtp_server, smtp_port)
         else:
-            context = ssl.create_default_context()
+            context = (
+                ssl.create_default_context() if self.verify else
+                ssl._create_unverified_context(protocol=ssl.PROTOCOL_TLS_CLIENT)
+            )
             if smtp_type == SMTPType.SSL:
                 server = SMTP_SSL(smtp_server, smtp_port, context=context)
             elif smtp_type == SMTPType.STARTTLS:
                 server = SMTP(smtp_server, smtp_port)
                 server.starttls(context=context)
-            if self.username is not None:
-                server.login(self.username, self.password.get_secret_value())
+        if self.username is not None:
+            if not self.verify or smtp_type == SMTPType.INSECURE:
+                get_run_logger().warning(
```

Comment:
Looks like tests are failing because `get_run_logger` cannot be used outside of a task or flow. Can you update this class to use `get_run_logger` when possible and fail back to `get_logger` otherwise?'